src/pypromice/tx/email_client/base_gmail_client.py

```diff
@@ -0,0 +1,43 @@
+from abc import ABC, abstractmethod
+from typing import Iterator, Tuple
+import email
+
+class BaseGmailClient(ABC):
```

Comment:
I think this should just be `BaseMailClient` as it is not fixed to Gmail mailboxes. I'll change this now.